ConfigSpace/io/pcs_new.py

```diff
@@ -0,0 +1,475 @@
+from ConfigSpace.hyperparameters import CategoricalHyperparameter, \
+    UniformIntegerHyperparameter, UniformFloatHyperparameter, \
+    NumericalHyperparameter, Constant, IntegerHyperparameter, \
+    NormalIntegerHyperparameter, NormalFloatHyperparameter, OrdinalHyperparameter
+from ConfigSpace.conditions import EqualsCondition, NotEqualsCondition,\
+    InCondition, AndConjunction, OrConjunction, ConditionComponent,\
+    GreaterThanCondition, LessThanCondition
+from ConfigSpace.forbidden import ForbiddenEqualsClause, \
+    ForbiddenAndConjunction, ForbiddenInClause, AbstractForbiddenComponent, MultipleValueForbiddenClause
+
+
+# Build pyparsing expressions for params
+pp_param_name = pyparsing.Word(pyparsing.alphanums + "_" + "-" + "@" + "." + ":" + ";" + "\\" + "/" + "?" + "!" +
+                               "$" + "%" + "&" + "*" + "+" + "<" + ">")
+pp_param_operation = pyparsing.Word("in" + "!=" + "==" + ">" + "<")
+pp_digits = "0123456789"
+pp_param_val = pp_param_name + pyparsing.Optional(pyparsing.OneOrMore("," + pp_param_name))
+pp_plusorminus = pyparsing.Literal('+') | pyparsing.Literal('-')
+pp_int = pyparsing.Combine(pyparsing.Optional(pp_plusorminus) + pyparsing.Word(pp_digits))
+pp_float = pyparsing.Combine(pyparsing.Optional(pp_plusorminus) + pyparsing.Optional(pp_int) + "." + pp_int)
+pp_eorE = pyparsing.Literal('e') | pyparsing.Literal('E')
+pp_param_type = pyparsing.Word("integer" + "real" + "categorical" + "ordinal")
+pp_floatorint = pp_float | pp_int
+pp_e_notation = pyparsing.Combine(pp_floatorint + pp_eorE + pp_int)
+pp_number = pp_e_notation | pp_float | pp_int
+pp_numberorname = pp_number | pp_param_name
+pp_log = pyparsing.Word("log")
+pp_connective = pyparsing.Word("||" + "&&")
+pp_choices = pp_param_name + pyparsing.Optional(pyparsing.OneOrMore("," + pp_param_name))
+pp_sequence = pp_param_name + pyparsing.Optional(pyparsing.OneOrMore("," + pp_param_name))
+pp_ord_param = pp_param_name + pp_param_type + "{" + pp_sequence + "}" + "[" + pp_param_name + "]"
+pp_cont_param = pp_param_name + pp_param_type + "[" + pp_number + "," + pp_number + "]" + "[" + pp_number + "]" + pyparsing.Optional(pp_log)
+pp_cat_param = pp_param_name + pp_param_type + "{" + pp_choices + "}" + "[" + pp_param_name + "]"
+pp_condition = pp_param_name + "|" + pp_param_name + pp_param_operation + pyparsing.Optional('{') + pp_param_val + pyparsing.Optional('}') + pyparsing.Optional(pp_connective +  pp_param_name + pp_param_operation + pp_param_val)
+pp_forbidden_clause = "{" + pp_param_name + "=" + pp_numberorname + \
+    pyparsing.Optional(pyparsing.OneOrMore("," + pp_param_name + "=" + pp_numberorname)) + "}"
+
+
+def build_categorical(param):
+    cat_template = "%s categorical {%s} [%s]"
+    return cat_template % (param.name,
+                           ", ".join([str(value) for value in param.choices]),
+                           str(param.default))
+
+def build_constant(param):
```

Comment:
I don't think there is a definition for a constant hyperparameter in the pcs definition, you need to encode it as a categorical.